digicampipe/scripts/pulse_shape.py

```diff
@@ -3,63 +3,204 @@
 Reconstruct the pulse template
 
 Usage:
-  pulse_shape.py [options] [FILE] [INPUT ...]
+  digicam-template [options] <input_files>...
 
 Options:
   -h --help               Show this screen.
-  --max_events=N          Maximum number of events to analyse
-  -o FILE --output=FILE.  Output file.
-  -i INPUT --input=INPUT. Input files.
-  -c --compute            Compute the data.
-  -f --fit                Fit.
-  -d --display            Display.
-  -v --debug              Enter the debug mode.
-  -p --pixel=<PIXEL>      Give a list of pixel IDs.
+  --output=PATH           outfile path, if not given, we just append ".h5" to
+                          the input file path
 '''
-import os
-from docopt import docopt
-from tqdm import tqdm
-from schema import Schema, And, Or, Use, SchemaError
 import numpy as np
-import matplotlib.pyplot as plt
-
+import h5py
 from digicampipe.io.event_stream import calibration_event_stream
-from digicampipe.io.containers_calib import CalibrationContainer
-
-
-def entry():
-
-    args = docopt(__doc__)
-
-    input_files = args['FILE']
-    max_events = args['--max_events']
-    max_events = max_events if max_events is None else str(max_events)
-
-    container = CalibrationContainer()
 
-    events = calibration_event_stream(input_files, telescope_id=1,
-                                      max_events=max_events)
-
-    for count, event in enumerate(events):
-
-        adc_samples = event.data.adc_samples
-
-        if count == 0:
-
-            pulse_template = np.zeros(adc_samples.shape)
-
-        pulse_template += adc_samples
-
-        plt.figure()
-        plt.plot(adc_samples[0])
-        plt.show()
+from tqdm import tqdm
+from docopt import docopt
 
-    pulse_template /= count
+import numba
+
+
+@numba.jit
+def estimate_arrival_time(adc, thr=0.5):
```

Comment:
I think this function could be included to this file : https://github.com/cta-sst-1m/digicampipe/blob/master/digicampipe/calib/camera/time.py

where here we try to collect all the time reconstruction methods. These methods :
1. take event containers (calibration events)
2. compute time
3. fill the dedicated field for time

One could rename it : `compute_time_from_leading_edge()`

Reply:
Thanks for the hint. You are right. However, I'd like to keep the function interface like this:
```python
def compute_time_from_leading_edge_using_linear(
    adc, # 2d, shape: (n_pixel, n_samples)
    threshold, # 0d, float
):
    # [...]
    return time # 1d, shape:(n_pixel,)
```

and maybe later provide a function which has this interface:
```python
def compute_time_from_leading_edge_using_linear(
    events
    threshold, # 0d, float
):
    # [...]
    yield event
```

The 1st version is simpler to unit test and at least in my opinion simpler to understand... the second version is simply needed in the container-ecosystem. So clearly both versions are needed. The question is just ... how should we do this? 

 - different function names for the different functions? 
 - or different modules? and the same function name?

Dunno ...